tools/bundler/index.js

```diff
@@ -29,14 +31,65 @@ console.log(`Using the following output directory: ${outputDirectory}`);
       const filePathToBundle = `file://${versionDir}/asyncapi.json`;
       const fileToBundle = await Bundler.get(filePathToBundle);
       const bundledSchema = await Bundler.bundle(fileToBundle);
+      modifyRefsandDefinitions(bundledSchema);
       bundledSchema.description = `!!Auto generated!! \n Do not manually edit. ${bundledSchema.description ?? ''}`;
       const outputFile = path.resolve(outputDirectory, `${version}.json`);
       console.log(`Writing the bundled file to: ${outputFile}`);
       await fs.promises.writeFile(outputFile, JSON.stringify(bundledSchema, null, 4));
     }catch(e) {
-      console.log(e);
+      throw new Error(e);
     }
   }
   console.log('done');
 })();
 
+
+/**
+ * we first update definitions from URL to normal names
+ * than update refs to point to new definitions, always inline never remote
+ */
+function modifyRefsandDefinitions(bundledSchema) {
+
+  //first we need to improve names of the definitions from URL to their names
+  for (const def of Object.keys(bundledSchema.definitions)) {
+    const newDefName = getDefinitionName(def);
+    
+    //creating copy of definition under new name so later definition stored under URL name can be removed
+    bundledSchema.definitions[newDefName] = bundledSchema.definitions[def];
+    delete bundledSchema.definitions[def]
+  }
+
+  traverse(bundledSchema, replaceRef);
+}
+
+/**
+ * by default schemas definition names are urls like http://asyncapi.com/definitions/2.4.0/parameters.json
+ * we need to get rid of URLs and use the last fragment as new definition name like `parameters`
+ */
+function getDefinitionName(def) {
+  if (def.startsWith('http://json-schema.org')) return JSON_SCHEMA_PROP_NAME;
```

Comment:
I know that we use only draft-07, however we should be hardcoded here only form draft-07 schema, so please change it to the:

```suggestion
  if (def.startsWith('http://json-schema.org/draft-07')) return JSON_SCHEMA_PROP_NAME;
```

Reply:
I accepted it initially but then reverted. It adds no value but causes complexity as in older spec versions we used draft 4. This would complicate code if I would like to have clear info about versions there

Reply:
Ok, thanks, I forgot about older versions.